pkg/codec/codec.go

```diff
@@ -0,0 +1,26 @@
+package codec
+
+// Encoder converts from textual representation to
+// bytes in the specified format
+type Encoder interface {
+	// Encode textual bytes to binary format
+	Encode(in []byte) ([]byte, error)
```

Comment:
let's talk about this interface. What exactly is our contract, that is shared for both avro, json ?

example of producing/encoding:

'some textual representation' => 'binary' 

i wonder, if the input to encode can be something else than []byte. i.e. string (stdin is technically []byte,so i think no, []byte is better).

a more radical approach could be: input is JSON (for example json.RawMessage). i.e. we say, if people want to use codecs, they input json for produce, or get (guaranteed) outputted JSON in consume.

i am not sure about this. for avro and json it makes sense, but for other formats? if i look at MessagePack, it's a binary format. But you can, with certain libs (https://github.com/ugorji/go), convert from/to JSON. Even cap'nproto can convert to/from JSON: https://capnproto.org/news/2017-05-01-capnproto-0.6-msvc-json-http-more.html.

We could design everything around this principle: kaf's users work with JSON, and codecs convert to/from the desired format.

again, not sure about this approach ,using JSON as the "common language".
i start liking this idea a lot, it could be just the "easy" developer experience i want to have for kaf.
let me hear your opinion!

Reply:
I like it! Especially because I didn't know about json.RawMessage before, I'm still getting familiar with golang's stdlib. 

I think it is a perfect fit for what we want to do!